src/include/Context.h

```diff
@@ -238,23 +214,12 @@ class C_ContextsBase : public ContextInstanceType {
   void add(ContextType* c) {
     contexts.push_back(c);
   }
-  void take(std::list<ContextType*>& ls) {
-    contexts.splice(contexts.end(), ls);
-  }
-  bool sync_complete(int r) override {
-    auto p = contexts.begin();
-    while (p != contexts.end()) {
-      if ((*p)->sync_complete(r)) {
-	p = contexts.erase(p);
-      } else {
-	++p;
-      }
-    }
-    if (contexts.empty()) {
-      delete this;
-      return true;
+  void take(Container& ls) {
+    if constexpr (std::is_same_v<Container, std::list<ContextType *>>) {
+      contexts.splice(contexts.end(), ls);
+    } else {
+      contexts.insert(contexts.end(), ls.begin(), ls.end());
```

Comment:
doesn't this need to contexts.clear() too?

Reply:
Yes, thanks!